src/server/services/Actions.ts

```diff
@@ -164,15 +178,20 @@ export default class Actions {
   ): Promise<DripResponse> {
     let dripTimeout: ReturnType<typeof rpcTimeout> | null = null;
     let result: DripResponse;
+    const parsedAmount = Number(amount);
+    const faucetBalance = this.getFaucetBalance();
 
     try {
       if (!this.account) throw new Error('account not ready');
 
-      const dripAmount = Number(amount) * 10 ** decimals;
-
+      if (faucetBalance && parsedAmount >= faucetBalance) {
```

Comment:
I think it should just be `parsedAmount >= faucetBalance`, otherwise it won't be caught when `faucetBalance` is `0`